compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8185,4 +8166,295 @@ private boolean isEndOfTypeList(SyntaxKind nextTokenKind) {
     private STNode parseTupleRestTypeDesc() {
         return STNodeFactory.createEmptyNode();
     }
+
+    /**
+     * Parse table constructor or query expression.
+     * <p>
+     * <code>
+     * table-constructor-or-query-expr := table-constructor-expr | query-expr
+     * <br/>
+     * table-constructor-expr := table [key-specifier] [ [row-list] ]
+     * <br/>
+     * query-expr := [query-construct-type] query-pipeline select-clause
+     * <br/>
+     * query-construct-type := table key-specifier | stream
+     * </code>
+     *
+     * @return Parsed node
+     */
+
+    private STNode parseTableConstructorOrQuery() {
+        startContext(ParserRuleContext.TABLE_CONSTRUCTOR_OR_QUERY_EXPRESSION);
+        return parseTableConstructorOrQuery(peek().kind);
```

Comment:
Can we end the context here, rather than in the respective child methods?
So that we can make sure the context is ended, no matter which branch it goes within the ` parseTableConstructorOrQuery`

Reply:
does that mean following change.
```
private STNode parseTableConstructorOrQuery() {
       startContext(ParserRuleContext.TABLE_CONSTRUCTOR_OR_QUERY_EXPRESSION);
       STNode expr = parseTableConstructorOrQuery(peek().kind);
       endContext();
       return expr;
```

Reply:
You mean this change right?
```java    
private STNode parseTableConstructorOrQuery() {
        startContext(ParserRuleContext.TABLE_CONSTRUCTOR_OR_QUERY_EXPRESSION);
        STNode tableOrQueryExpr = parseTableConstructorOrQuery(peek().kind);
        endContext();
        return tableOrQueryExpr;
    }
```

Reply:
yes.